disnake/interactions/base.py

```diff
@@ -649,26 +670,60 @@ def is_done(self) -> bool:
         """
         return self._responded
 
-    async def defer(self, *, ephemeral: bool = False, with_message: bool = False) -> None:
+    @overload
+    async def defer(
+        self,
+        *,
+        with_message: Literal[True],
+        ephemeral: bool = ...,
+    ):
+        ...
+
+    @overload
+    async def defer(
+        self,
+        *,
+        ephemeral: bool = ...,
+    ):
+        ...
+
+    @overload
+    async def defer(
+        self,
+        *,
+        with_message: Literal[False],
+    ):
+        ...
+
+    async def defer(
+        self,
+        *,
+        with_message: bool = MISSING,
+        ephemeral: bool = MISSING,
+    ) -> None:
         """|coro|
 
         Defers the interaction response.
 
         This is typically used when the interaction is acknowledged
         and a secondary action will be done later.
 
+        .. versionchanged:: 2.5
+
+            ``ephemeral`` now implies ``with_message`` if it is not already set.
+
         Parameters
         ----------
-        ephemeral: :class:`bool`
-            Whether the deferred message will eventually be ephemeral.
-            This applies to interactions of type :attr:`InteractionType.application_command` and :attr:`InteractionType.modal_submit`
-            or when the ``with_message`` parameter is ``True``.
         with_message: :class:`bool`
             Whether the response will be a message with thinking state (bot is thinking...).
-            This only applies to interactions of type :attr:`InteractionType.component`.
 
             .. versionadded:: 2.4
 
+        ephemeral: :class:`bool`
+            Whether the deferred message will eventually be ephemeral.
+
+            .. note:: This automatically implies ``with_message`` is True.
+
```

Comment:
Given suppress_embeds doesn't work, but the api says it should... we should note that here, somewhere, as part of the documentation. IMO _best_ practice would be to have it but nonfunctional, and labeled as such, but if you want to have it unavailable that's functional as well. However, this should be documented

Reply:
I agree that would be best practice *if* the team behind the api had a track record of updating the api to match the docs rather than update the docs to match the api.

Reply:
deferred to later pr